sklearn/impute/_iterative.py

```diff
@@ -581,6 +582,36 @@ def fit_transform(self, X, y=None):
             self.n_iter_ = 0
             return super()._concatenate_indicator(Xt, X_indicator)
 
+        # Convert min and max values to float arrays if array-like
+        # Else (i.e, if scalar) broadcast it to an array of shape=n_features
+        self._min_value = (
+            as_float_array(self.min_value, force_all_finite=False)
+            if _is_arraylike(self.min_value)
+            else as_float_array(np.full(X.shape[1], self.min_value),
+                                force_all_finite=False)
+        )
+        self._max_value = (
+            as_float_array(self.max_value, force_all_finite=False)
+            if _is_arraylike(self.max_value)
+            else as_float_array(np.full(X.shape[1], self.max_value),
+                                force_all_finite=False)
+        )
+        # Fill in any np.nan in min and max value arrays
+        # with -np.inf and np.inf respectively
+        self._min_value[np.isnan(self._min_value)] = -np.inf
+        self._max_value[np.isnan(self._max_value)] = np.inf
+
+        # Raise a ValueError if min value for any feature is >= its max value
+        if not np.all(np.greater(self._max_value, self._min_value)):
+            raise ValueError(
+                "One (or more) features have min_value >= max_value")
+        # Raise a ValueError if _min_value and _max_value
+        # are not the same shape as n_features
+        if not (self._min_value.shape[0] == X.shape[1] and
```

Comment:
This check should be done before the previous one.

Reply:
Done!